### ISSUE TITLE
Enabling encryption with VXLAN causes communication failures between pods on different nodes

### ISSUE THREAD

**Issue by iczero** [cilium/cilium#15756]
## Bug report

When using Cilium with VXLAN and encryption across multiple nodes, pods on those nodes will lose connection with each other. This only occurs if encryption is enabled. If Cilium is reinstalled on the same cluster with encryption disabled, this issue does not happen.

**General Information**

- Cilium version: 1.9.5
- Kernel version: 5.8.0-49-generic (Ubuntu 20.10)
- Kernel version of VMs: 5.4.0-72-generic (Ubuntu 20.04)
- Kubernetes version: 1.21.0
- Link to relevant artifacts: see below
- Generate and upload a system zip: see attached
  [cilium-sysdump-20210416-203732.zip](https://github.com/cilium/cilium/files/6328616/cilium-sysdump-20210416-203732.zip)

3 nodes exist in the cluster: `desktop`, which hosts 2 VMs on QEMU/KVM on libvirt, `kubernetes-node-1` and `kubernetes-node-2`. VM networking happens over the libvirt bridge interface. The desktop is on Ubuntu 20.10 and the VMs are on Ubuntu 20.04. The control plane runs on `desktop`. Firewalls were entirely disabled.

`kubeadm init` config file:
```yaml
apiVersion: kubeadm.k8s.io/v1beta2
kind: InitConfiguration
localAPIEndpoint:
  advertiseAddress: 172.19.24.15
  bindPort: 6443
nodeRegistration:
  ignorePreflightErrors:
  - Swap
---
apiVersion: kubeadm.k8s.io/v1beta2
kind: ClusterConfiguration
clusterName: kubernetes-desktop
etcd:
  local:
    # provided due to etcd write issues on btrfs
    dataDir: /mnt/datapool/var/lib/etcd
networking:
  podSubnet: "10.32.0.0/16,fd54:b4af:fa00::/48"
  serviceSubnet: "10.33.0.0/16,fd54:b4af:fa01::/108"
  dnsDomain: "cluster.local"
apiServer:
  certSANs:
  - "iczero-desktop"
---
apiVersion: kubelet.config.k8s.io/v1beta1
kind: KubeletConfiguration
failSwapOn: false
```

`kubeadm join` config file:
```yaml
apiVersion: kubeadm.k8s.io/v1beta2
kind: JoinConfiguration
nodeRegistration:
  ignorePreflightErrors:
  - Swap
discovery:
  bootstrapToken:
    # replace these values with the corresponding values from the bootstrap token
    apiServerEndpoint: 172.19.24.15:6443
    caCertHashes:
    - "sha256:HASH"
    token: "TOKEN"
```

cilium installation command:
```
helm install cilium cilium/cilium --namespace=kube-system \
  --set ipv6.enabled=true \
  --set encryption.enabled=true \
  --set ipam.mode=cluster-pool \
  --set ipam.operator.clusterPoolIPv4PodCIDR=10.32.0.0/16 \
  --set ipam.operator.clusterPoolIPv6PodCIDR=fd54:b4af:fa00::/48 \
  --set ipam.operator.clusterPoolIPv4MaskSize=24 \
  --set ipam.operator.clusterPoolIPv6MaskSize=64 \
  --set operator.replicas=1 \
  --set kubeProxyReplacement=disabled
```

**How to reproduce the issue**

1. Bootstrap kubernetes with kubeadm with the provided configs (change IPs if necessary)
2. Install cilium with provided helm command
3. Join the other two nodes with kubeadm
4. Wait, and perhaps ping some nodes
5. Run something like `kubectl -n kube-system get pods -l k8s-app=cilium | tail -n +2 | cut -d ' ' -f 1 | sort | parallel -j1 kubectl -n kube-system exec '{}' -c cilium-agent -- cilium-health status` and observe failures, or just ping those nodes manually

**Other observations**

 - Running `ip xfrm state flush` on all nodes sometimes makes the issue better for a while
 - Restarting all cilium pods with `kubectl -n kube-system rollout restart daemonset cilium` will cause different nodes to experience failures
 - Sometimes `ip xfrm policy` will have the wrong IP
 - Both `XfrmInStateProtoError` and `XfrmInStateSeqError` in `/proc/net/xfrm_stat` increment when pinging over an affected connection
 - Wireshark will occasionally report "wrong sequence number" on ESP packets over `cilium_vxlan`

Note: I have already asked about this in the Slack channel and was directed to open a bug report here. Through that discussion it was determined that fragmentation issues were most likely not the cause of this issue. I have attempted to use `cilium/mtu-update` to drastically reduce the mtu and raised the mtu on the bridge interface but none of those helped.

**Comment by pchaigno** [comment]
This is a known bug, introduced by https://github.com/cilium/cilium/pull/15039. The revert for that commit will be included in v1.9.6.

**Comment by iczero** [comment]
Can confirm this is no longer an issue in v1.9.6. Thanks a lot!
